ignitionrobotics/ign-gazebo
# Add support for configuring point size in Visualize Lidar GUI plugin

# 🎉 New feature

depends on https://github.com/ignitionrobotics/ign-rendering/pull/392

## Summary

Added a spin box for configuring size of lidar point visualization in `Visualize Lidar` plugin. Note it only affects `Point` visualization type. Works for `ogre` and `ogre2` render engine.

## Test it

1. Launch an example world with gpu ray sensor, e.g.:

    ```
    ign gazebo -v 4 gpu_lidar_sensor.sdf
    ```

1. Open the `Visualize Lidar` GUI plugin by going to the top right menu (3 vertical dots), scroll down and click on `Visualize Lidar`.

1. Unpause the simulation by hitting the play button
1. Refresh the topic drop down list and select the `/lidar` topic
1. Select `Points` Visual Type to see the default point visualization with size `1`
1. Change `Point Size` to `5` and hit Enter to see larger points.

![lidar_point_size](https://user-images.githubusercontent.com/4000684/132771657-f5599431-e57d-4527-973e-282c40eab9e4.png)

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge**
